Fix racy install of cmake package config files

Lasander opened the pull request:
This is to fix an issue that we observed in an Yocto linux environment when building rapidjson. The installation included cmake package configuration files that pointed to the absolution paths in the original build environment.

Upon further investigation the issue turned out to be a racy installation of such files within rapidjson. This patch should fix the issue.

--

**Manual testing**

Before the change the install target would output:
```
-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfig.cmake
-- Up-to-date: /.../install/lib/cmake/RapidJSON/RapidJSONConfig.cmake
-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfigVersion.cmake
```
showing the attempt to install the same file twice.

After the change the install target would output only:
```
-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfig.cmake
-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfigVersion.cmake
```

--

The content of the installed file is as expected. References to include directory are relative to the package config file location:
```
# Compute paths
get_filename_component(RapidJSON_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)

set( RapidJSON_INCLUDE_DIR  "${RapidJSON_CMAKE_DIR}/../../../include" )
set( RapidJSON_INCLUDE_DIRS  "${RapidJSON_CMAKE_DIR}/../../../include" )
message(STATUS "RapidJSON found. Headers: ${RapidJSON_INCLUDE_DIRS}")
```

Lasander committed:
Install also RapidJSONConfigVersion.cmake for the install tree

Configure the RapidJSONConfigVersion.cmake to the location reserved for
the files for the install tree. Install it along-side the file
RapidJSONConfig.cmake from the same location.

Signed-off-by: Lasse Lopperi <lasse.lopperi@iki.fi>

tencent-adm commented:
[![CLA assistant check](https://cla.opensource.tencent.com/pull/badge/not_signed)](https://cla.opensource.tencent.com/Tencent/rapidjson?pullRequest=2167) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/rapidjson?pullRequest=2167) before we can accept your contribution.<br/><hr/>**Lasse Lopperi** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/rapidjson?pullRequest=2167) it.</sub>

Lasander committed:
Don't install file meant for use within local build

Remove the installation of RapidJSONConfig/ConfigVersion.cmake files
that were configured to the root of the build directory. According to
the comments, these are meant for somekind of usage within the local
build tree. The content of RapidJSONConfig.cmake refers to local
absolute directories, so installing it doesn't make sense.

Also, the previous attempts to install it resulted in a race with the
proper RapidJSONConfig.cmake file. This would be visible when running
the install target:

-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfig.cmake
-- Up-to-date: /.../install/lib/cmake/RapidJSON/RapidJSONConfig.cmake

In the worst case the file aimed for local environment would be
installed first resulting in a non-functioning install.

The "COMPONENT dev" is kept in the config file install for consistency
with other installation commands. As there is no EXCLUDE_FROM_ALL, the
"dev" component will be part of the default install target anyways.

Signed-off-by: Lasse Lopperi <lasse.lopperi@iki.fi>

Lasander commented:
> [![CLA assistant check](https://camo.githubusercontent.com/a294d8f9ee1b945c56ac70d174214538480551cba2ce1193453ca7f0f1eb92a9/68747470733a2f2f636c612e6f70656e736f757263652e74656e63656e742e636f6d2f70756c6c2f62616467652f6e6f745f7369676e6564)](https://cla.opensource.tencent.com/Tencent/rapidjson?pullRequest=2167)
> Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/rapidjson?pullRequest=2167) before we can accept your contribution.
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/rapidjson?pullRequest=2167) it.

Getting a "504 Gateway Time-out" for the CLA signing - so cannot complete that.

Lasander commented:
Based on the description, this would fix: https://github.com/Tencent/rapidjson/issues/1800

Lasander commented:
> Getting a "504 Gateway Time-out" for the CLA signing - so cannot complete that.

I think I got through.. so should be in progress now.
